components/library/tdb/Makefile

```diff
@@ -11,40 +11,50 @@
 
 #
 # Copyright 2017 Alexander Pyhalov
+# Copyright (c) 2022, Friedrich Kink
 #
 
+BUILD_BITS =                    64
+BUILD_STYLE =                   waf
+
 include ../../../make-rules/shared-macros.mk
 
 COMPONENT_NAME= tdb
-COMPONENT_VERSION= 1.3.12
+COMPONENT_VERSION= 1.4.7
 COMPONENT_REVISION= 1
 COMPONENT_SUMMARY= Trivial Database Library
 COMPONENT_SRC= $(COMPONENT_NAME)-$(COMPONENT_VERSION)
 COMPONENT_ARCHIVE= $(COMPONENT_SRC).tar.gz
-COMPONENT_ARCHIVE_HASH= \
-  sha256:60134e32253cac8e2efe5e0185d20123c208bcf6ad15edf2f50d80daadf8c348
-COMPONENT_ARCHIVE_URL= \
-  https://download.samba.org/pub/tdb/$(COMPONENT_ARCHIVE)
+COMPONENT_ARCHIVE_HASH= sha256:a4fb168def533f31ff2c07f7d9844bb3131e6799f094ebe77d0380adc987c20e
+COMPONENT_ARCHIVE_URL= https://download.samba.org/pub/tdb/$(COMPONENT_ARCHIVE)
 COMPONENT_PROJECT_URL = https://wiki.samba.org/index.php/TDB
 COMPONENT_FMRI = library/tdb
 COMPONENT_CLASSIFICATION = Development/Databases
 # Binaries are GPLv3, but we don't ship them
 COMPONENT_LICENSE = LGPLv3
 
-include $(WS_MAKE_RULES)/prep.mk
-include $(WS_MAKE_RULES)/waf.mk
-include $(WS_MAKE_RULES)/ips.mk
+include $(WS_MAKE_RULES)/common.mk
+
+PYTHON_VERSION = 3.9
+
+COMPONENT_PRE_CONFIGURE_ACTION += ( \
+	$(CLONEY) $(SOURCE_DIR) $(@D); \
+	cd $(@D); \
+	);
+
+CFLAGS +=       -D_POSIX_PTHREAD_SEMANTICS
+CPPFLAGS +=     $(CPP_LARGEFILES) $(CPP_XPG6MODE)
+# Flag is not picked up for lib/replace
+LDFLAGS +=      $(LD_SSP)
 
-WAF = $(PYTHON) buildtools/bin/waf
+WAF = PYTHONHASHSEED=1 $(PYTHON) ./buildtools/bin/waf
 
 # Waf doesn't like variables passing via args, so we rewrite CONFIGURE_OPTIONS
 CONFIGURE_OPTIONS = --prefix=$(CONFIGURE_PREFIX)
-CONFIGURE_OPTIONS += --bindir=$(CONFIGURE_BINDIR.$(BITS))
-CONFIGURE_OPTIONS += --sbindir=$(CONFIGURE_SBINDIR.$(BITS))
+CONFIGURE_OPTIONS += --bindir=$(CONFIGURE_BINDIR)
+CONFIGURE_OPTIONS += --sbindir=$(CONFIGURE_SBINDIR)
 CONFIGURE_OPTIONS += --mandir=$(CONFIGURE_MANDIR)
-CONFIGURE_OPTIONS += --libdir=$(CONFIGURE_LIBDIR.$(BITS))
-CONFIGURE_OPTIONS += --without-gettext
-CONFIGURE_OPTIONS += --disable-python
+CONFIGURE_OPTIONS += --libdir=$(CONFIGURE_LIBDIR)
 CONFIGURE_OPTIONS += --disable-rpath
 
 COMPONENT_TEST_MASTER = $(COMPONENT_TEST_RESULTS_DIR)/results-all.master
```

Comment:
COMPONENT_TEST_MASTER = results-all.master is the default now. You can remove this line.